rocketpy/mathutils/function.py

```diff
@@ -2197,10 +2215,56 @@ def __sub__(self, other):
         result : Function
             A Function object which gives the result of self(x)-other(x).
         """
-        try:
-            return self + (-other)
-        except TypeError:
-            return Function(lambda x: (self.get_value_opt(x) - other(x)))
+        other_is_func = isinstance(other, Function)
+        other_array = other._source_type is SourceType.ARRAY if other_is_func else False
```

Comment:
maybe? 

```suggestion
        other_is_array = other._source_type is SourceType.ARRAY if other_is_func else False
```

Reply:
Sure! Done in [ae07fcb](https://github.com/RocketPy-Team/RocketPy/pull/810/commits/ae07fcb1cf3083f5c5050ab945c3e42511ecca47).